dask_expr/_collection.py

```diff
@@ -721,6 +725,10 @@ def shuffle(
         --------
         >>> df = df.shuffle(df.columns[0])  # doctest: +SKIP
         """
+        if on is no_default and not index_shuffle:
+            raise TypeError("Have to pass on or set index_shuffle to True")
+        elif on is not no_default and index_shuffle:
+            raise TypeError("Can't pass on and set index_shuffle to True")
```

Comment:
I'm a bit confused. IIUC, this states that 

IF `on is no_default`, THEN `index_shuffle` has to be `True`. 
OTHERWISE, (i.e, `on is not no_default`), THEN `index_shuffle` has to be `False`.

If that's correct, why not simply infer the value via `index_shuffle = on is no_default`?

Reply:
Too implicit for my liking, especially since this is a new keyword

Reply:
From an offline conversation: I'd suggest renaming `shuffle_index` to `on_index` which should make the relationship between `on` and `on_index` more explicit.

Reply:
I'd also adjust the error message to something that makes the relationship clearer, e.g., 

"Must shuffle on either columns or the index; currently shuffling on neither. Pass column(s) to `on` or set `on_index` to `True`."

"Cannot shuffle on both columns and the index. Do not pass column(s) to `on` or set `on_index` to `False`.